agent/taskresource/volume/dockervolume.go

```diff
@@ -71,40 +91,124 @@ func NewVolumeResource(name string,
 		},
 		client: client,
 	}
+	v.initializeResourceStatusToTransitionFunction(scope)
+	return v
+}
+
+func (vol *VolumeResource) initializeResourceStatusToTransitionFunction(scope string) {
```

Comment:
Related to my previous comment, `initStatusToTransitions` is a smaller and friendlier name for this.